From 714fe9a352e4c8ae6cb20cbefe4eff3c14c6f132 Mon Sep 17 00:00:00 2001
From: Jyotiraditya <jyotiraditya@aospa.co>
Date: Tue, 19 May 2020 16:55:28 +0200
Subject: [PATCH 5/7] Trebuchet: Add Smartspace Widget.

Since we removed QSB on top it looks empty at the fist glance after boot.
So, add "At a glance" widget provided by Google to the default workspace,
this allows the user to move it around or remove it as per their choice.

Change-Id: I4c2d7007595ceedca454e31ed751e75b06093c8d
(cherry picked from commit 6d962cea2cd351142d92dc71e229bb76b78086a2)

Launcher3: Add smartspace widget to 3x3 and 6x5 grids

Change-Id: Ib1426c9e52e1ebc065bad41b04a7b7eb8c92531c
Signed-off-by: Omkar Chandorkar <gotenksIN@aospa.co>
---
 res/xml/default_workspace_2x2.xml | 10 ++++++++++
 res/xml/default_workspace_3x3.xml | 10 ++++++++++
 res/xml/default_workspace_4x5.xml | 10 ++++++++++
 res/xml/default_workspace_5x5.xml | 10 ++++++++++
 res/xml/default_workspace_5x6.xml | 10 ++++++++++
 res/xml/default_workspace_5x7.xml | 10 ++++++++++
 res/xml/default_workspace_6x5.xml | 10 ++++++++++
 res/xml/default_workspace_6x6.xml | 10 ++++++++++
 8 files changed, 80 insertions(+)

diff --git a/res/xml/default_workspace_2x2.xml b/res/xml/default_workspace_2x2.xml
index 51d9e07005..c0bac5bef3 100644
--- a/res/xml/default_workspace_2x2.xml
+++ b/res/xml/default_workspace_2x2.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="4"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Messaging, Dialer -->
 
diff --git a/res/xml/default_workspace_3x3.xml b/res/xml/default_workspace_3x3.xml
index 31376e1d5e..0a22f987b5 100644
--- a/res/xml/default_workspace_3x3.xml
+++ b/res/xml/default_workspace_3x3.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="3"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Messaging, [All Apps], Dialer -->
 
diff --git a/res/xml/default_workspace_4x5.xml b/res/xml/default_workspace_4x5.xml
index 05c64acfd0..87285e7fc4 100644
--- a/res/xml/default_workspace_4x5.xml
+++ b/res/xml/default_workspace_4x5.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="4"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Dialer, Messaging, Browser, Camera -->
     <resolve
diff --git a/res/xml/default_workspace_5x5.xml b/res/xml/default_workspace_5x5.xml
index dba360132f..e2484b41d6 100644
--- a/res/xml/default_workspace_5x5.xml
+++ b/res/xml/default_workspace_5x5.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="5"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Dialer, Messaging, Contacts, Browser, Camera -->
     <resolve
diff --git a/res/xml/default_workspace_5x6.xml b/res/xml/default_workspace_5x6.xml
index 7e3257d541..ed0d8c2035 100644
--- a/res/xml/default_workspace_5x6.xml
+++ b/res/xml/default_workspace_5x6.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="5"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Dialer, Messaging, [Maps/Music], Browser, Camera -->
     <resolve
diff --git a/res/xml/default_workspace_5x7.xml b/res/xml/default_workspace_5x7.xml
index fe2362f635..2a0e2278ac 100644
--- a/res/xml/default_workspace_5x7.xml
+++ b/res/xml/default_workspace_5x7.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="5"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Dialer, Messaging, [Maps/Music], Browser, Camera -->
     <resolve
diff --git a/res/xml/default_workspace_6x5.xml b/res/xml/default_workspace_6x5.xml
index 39aaa306b3..8601f48460 100644
--- a/res/xml/default_workspace_6x5.xml
+++ b/res/xml/default_workspace_6x5.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="2"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Contacts, Calendar, Gallery, Music, Browser, Camera -->
     <resolve
diff --git a/res/xml/default_workspace_6x6.xml b/res/xml/default_workspace_6x6.xml
index f6ab9292db..e9f4e751da 100644
--- a/res/xml/default_workspace_6x6.xml
+++ b/res/xml/default_workspace_6x6.xml
@@ -16,6 +16,16 @@
 
 <favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
 
+    <!-- Smartspace Widget -->
+    <appwidget
+        launcher:screen="0"
+        launcher:x="0"
+        launcher:y="0"
+        launcher:spanX="5"
+        launcher:spanY="1"
+        launcher:packageName="com.google.android.googlequicksearchbox"
+        launcher:className="com.google.android.apps.gsa.staticplugins.smartspace.widget.SmartspaceWidgetProvider" />
+
     <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
     <!-- Dialer, Messaging, [Maps/Music], Browser, Camera -->
     <resolve
-- 
2.34.1

